lib/livebook_web/plugs/user_plug.ex

```diff
@@ -78,4 +80,20 @@ defmodule LivebookWeb.UserPlug do
     user_data = conn.cookies["lb_user_data"] |> Base.decode64!() |> Jason.decode!()
     put_session(conn, :user_data, user_data)
   end
+
+  defp identity_provider_module() do
+    livebook_teams_zta? =
+      case Livebook.Hubs.get_default_hub() do
+        %{id: "team-" <> _ = id} -> Livebook.Hubs.TeamClient.livebook_teams_zta?(id)
+        _ -> false
+      end
+
+    if livebook_teams_zta? do
+      Livebook.ZTA.LivebookTeams.start_link(name: @zta_name)
```

Comment:
We shouldn't link to the request process. In general we still want to start it under the supervision tree.

And `Livebook.Config.identity_provider` should return the teams one, otherwise we can run into bugs if we rely on it elsewhere. We can do `Application.put_env(:livebook, :identity_provider, ...)` on boot, after inserting the online hub (but only if the configured provider is pass through). If we do it, it should start under the supervision tree with the current logic we have. Does that work?

Reply:
Ah, the issue is that we would need to make `livebook_teams_zta?` as part of the ZTA, and if it's not enabled, we effectively want to fall back to session passthrough auth.

Reply:
I think if `livebook_teams_zta?` is disabled we could just return `{conn, %{}}` from authenticate, the same way that `PassThrough` does, so it's effectively disabled.

Reply:
Oh, `start_link` doesn't actually start anything :D But regardless, all the comments are still valid, we should not special case teams here :)

Reply:
This code should be part of the teams ZTA.